alacritty/src/display/hint.rs

```diff
@@ -33,6 +35,27 @@ pub struct HintState {
     keys: Vec<char>,
 }
 
+#[derive(Debug, Clone)]
+pub enum HintMatchLocation {
+    Regex { bounds: Match },
+    Hyperlink { hyperlink: Hyperlink, bounds: Match },
+}
+
+impl HintMatchLocation {
+    pub fn bounds(&self) -> Match {
+        match self {
+            Self::Regex { bounds } | Self::Hyperlink { bounds, .. } => bounds.clone(),
+        }
+    }
+
+    fn hyperlink(&self) -> Option<&Hyperlink> {
+        match self {
+            Self::Hyperlink { hyperlink, .. } => Some(hyperlink),
+            _ => None,
+        }
+    }
```

Comment:
This method seems a little strange. Not the end of the world but I wouldn't expect a method exclusive to a single of the enum variants on the entire enum that just returns None otherwise. Seems like at that point you'd rather just do the logic externally?